lib/src/utilities.dart

```diff
@@ -237,32 +265,110 @@ class Utils {
+    return itemInStorage;
+  }
+
+  static Future<String?> getTokenForHeaderAuth(TokenType tokenType) async {
+    String name = tokenType.getStorageName();
+    return await getFromStorage(name);
+  }
+
+  static Future<http.BaseRequest> setAuthorizationHeaderIfRequired(
+      http.BaseRequest mutableRequest,
+      {bool addRefreshToken = false}) async {
+    String? accessToken = await Utils.getTokenForHeaderAuth(TokenType.ACCESS);
+    String? refreshToken = await Utils.getTokenForHeaderAuth(TokenType.REFRESH);
+
+    if (accessToken != null && refreshToken != null) {
+      if (mutableRequest.headers["Authorization"] != null ||
+          mutableRequest.headers["authorization"] != null) {
+        //  no-op
+      } else {
+        String tokenToAdd = addRefreshToken ? refreshToken : accessToken;
+        mutableRequest.headers["Authorization"] = "Bearer $tokenToAdd";
+      }
+    }
+    return mutableRequest;
+  }
+
+  static Future<http.Request> setAuthorizationHeaderIfRequiredForRequestObject(
+      http.Request mutableRequest,
+      {bool addRefreshToken = false}) async {
+    String? accessToken = await Utils.getTokenForHeaderAuth(TokenType.ACCESS);
+    String? refreshToken = await Utils.getTokenForHeaderAuth(TokenType.REFRESH);
+
+    if (accessToken != null && refreshToken != null) {
+      if (mutableRequest.headers["Authorization"] != null) {
+        //  no-op
+      } else {
+        String tokenToAdd = addRefreshToken ? refreshToken : accessToken;
+        mutableRequest.headers["Authorization"] = "Bearer $tokenToAdd";
+      }
+    }
+    return mutableRequest;
+  }
+
+  static void setToken(TokenType tokenType, String value) async {
+    String name = tokenType.getStorageName();
+    return await SuperTokensUtils.storeInStorage(name, value);
+  }
+
+  static Future<void> saveTokenFromHeaders(
+      http.StreamedResponse response) async {
+    Map<String, String> headers = response.headers;
+
+    if (headers[frontTokenHeaderKey] != null) {
+      await FrontToken.setItem(headers[frontTokenHeaderKey]!);
+    }
+
+    if (headers[antiCSRFHeaderKey] != null) {
+      LocalSessionState localSessionState =
+          await SuperTokensUtils.getLocalSessionState();
+
+      await AntiCSRF.setToken(
+          headers[antiCSRFHeaderKey]!, localSessionState.lastAccessTokenUpdate);
+    }
+
+    if (headers[ACCESS_TOKEN_NAME] != null) {
+      setToken(TokenType.ACCESS, headers[ACCESS_TOKEN_NAME]!);
+    }
+
+    if (headers[REFRESH_TOKEN_NAME] != null) {
+      setToken(TokenType.REFRESH, headers[REFRESH_TOKEN_NAME]!);
+    }
+  }
 }
 
 class NormalisedInputType {
   late String apiDomain;
   late String? apiBasePath;
   late int sessionExpiredStatusCode = 401;
-  late String? cookieDomain;
-  late String? userDefaultdSuiteName;
+  late String? sessionTokenBackendDomain;
+  late SuperTokensTokenTransferMethod tokenTransferMethod;
+  late String? userDefaultSuiteName;
```

Comment:
Is this used anywhere?

Reply:
nope, had copied over from iOS
Somehow missed it in previous PRs